datastore/updater.go

```diff
@@ -19,6 +22,9 @@ type Updater interface {
 	// vulnerabilities, and ensures vulnerabilities from previous updates are
 	// not queried by clients.
 	UpdateVulnerabilities(ctx context.Context, updater string, fingerprint driver.Fingerprint, vulns []*claircore.Vulnerability) (uuid.UUID, error)
+	// UpdateVulnerabilitiesIter performs the same operation as
+	// UpdateVulnerabilities, but accepting an iterator function.
+	UpdateVulnerabilitiesIter(ctx context.Context, updater string, fingerprint driver.Fingerprint, vulnIter VulnerabilityIter) (uuid.UUID, error)
```

Comment:
> 	// The passed iterator must be able to iterate over the sequence multiple times.

That's not true for `UpdateVulnerabilitiesIter`. It would be for a potential `DeltaUpdateVulnerabilitiesIter`, but this patch does not aim to support that.